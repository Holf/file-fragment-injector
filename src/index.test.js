const fs = require("fs");
const { injectFileFragments } = require("./index.js");

const targetFilePath = "./testAssets/generated/destFile.html";

describe("File fragment injection", () => {
    it("should inject file fragments", () => {
        injectFileFragments("./testAssets/sourceFile.html", targetFilePath);

        const generatedFile = fs.readFileSync(targetFilePath, "utf8");

        expect(generatedFile).toMatchSnapshot();
    });
});
