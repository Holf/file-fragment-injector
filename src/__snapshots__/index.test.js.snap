// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File fragment injection should 1`] = `
"<!doctype html>
<html lang="en">
    <head>
        <script>
            // The following was generated by a C-style comment.
            console.log("Hello from fragmentFile1.js");

        </script>
        <meta charset="UTF-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <title>Document</title>
    </head>
    <body>
        <!-- The following was generated by an HTML-style comment. -->
        <div>This is some content from fragmentFile2.html</div>

        <div>
            <pre>
The following was injected by means of a Shell-style comment.
console.log("Hello from fragmentFile1.js");


The following was injected by means of a Powershell comment.
<div>This is some content from fragmentFile2.html</div>

      </pre
            >
        </div>
        <script>
            // Here are some comments which should not be replaced by injection:

            // Missing the leading %:
            //   # FRAGMENT_PATH:./fragmentFile1.js %#

            // Missing the trailing %:
            //   #% FRAGMENT_PATH:./fragmentFile1.js #

            // Typo in FRAGMENT_PATH:
            //   #% FRAGMNT_PATH:./fragmentFile1.js %#
        </script>
    </body>
</html>
"
`;
